dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelAIPluginExtensions.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+public static class KernelAIPluginExtensions
+{
+    /// <summary>
+    /// Imports an AI plugin that is exposed as an OpenAPI v3 endpoint or through OpenAI's ChatGPT format.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="filePath">The file path to the AI Plugin</param>
+    /// <param name="executionParameters">Skill execution parameters.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportAIPluginAsync(
+        this IKernel kernel,
+        string skillName,
+        string filePath,
+        OpenApiSkillExecutionParameters? executionParameters = null,
```

Comment:
I think we'll need an (optional) `HttpClient? httpClient = null` arg here.
In the public methods here, we always load the HttpClientProvider one, but there's a high likelihood that some of these AI plugin APIs will require custom bearer tokens and such. We should allow optional passing, and if null, fall back to what you have below.

@SergeyMenshykh to confirm

Reply:
I believe this is passed in via the OpenApiSkillExecutionParameters: https://github.com/microsoft/semantic-kernel/blob/adc59234aa6d1f91fe65376b3a9a5074b4f0dae9/dotnet/src/Skills/Skills.OpenAPI/Extensions/OpenApiSkillExecutionParameters.cs#L19

Reply:
@SergeyMenshykh anything to add here?